app/pages/nominees/IsOrganisationNomineeNinoPage.scala

```diff
@@ -16,12 +16,22 @@
 
 package pages.nominees
 
+import models.UserAnswers
 import pages.QuestionPage
 import play.api.libs.json.JsPath
 
+import scala.util.Try
+
 case object IsOrganisationNomineeNinoPage extends QuestionPage[Boolean] {
 
   override def path: JsPath = NomineeOrganisationIdPage.path \ toString
 
   override def toString: String = "isOrganisationNino"
+
+  override def cleanup(value: Option[Boolean], userAnswers: UserAnswers): Try[UserAnswers] =
+    value match {
+      case Some(true)  => userAnswers.remove(OrganisationAuthorisedPersonPassportPage)
+      case Some(false)  => userAnswers.remove(OrganisationAuthorisedPersonNinoPage)
+      case _ => super.cleanup(value, userAnswers)
```

Comment:
correct me if I'm wrong, but when cleanup is called, value is always a Some (as if the page is a None, this function won't actually be called) - so you can just replace lines 34 and 35 with a `Case _ => userAnswers.remove(OrganisationAuthorisedPersonNinoPage)`

Reply:
modified